src/Compilers/Core/Portable/DiagnosticAnalyzer/DiagnosticAnalysisContext.cs

```diff
@@ -1453,4 +1454,96 @@ public void ReportDiagnostic(Diagnostic diagnostic)
         /// </summary>
         public ControlFlowGraph GetControlFlowGraph() => DiagnosticAnalysisContextHelpers.GetControlFlowGraph(Operation, _getControlFlowGraphOpt, _cancellationToken);
     }
+
+    /// <summary>
+    /// Context for suppressing analyzer and/or compiler non-error diagnostics reported for the compilation.
+    /// </summary>
+    public struct SuppressionAnalysisContext
+    {
+        private readonly Action<Suppression> _addSuppression;
+        private readonly Func<SuppressionDescriptor, bool> _isSupportedSuppressionDescriptor;
+        private readonly Func<SyntaxTree, SemanticModel> _getSemanticModel;
+
+        /// <summary>
+        /// Suppress analyzer and/or compiler non-error diagnostics reported for the compilation.
+        /// This may be a subset of the full set of reported diagnostics, as an optimization for
+        /// supporting incremental and partial analysis scenarios.
+        /// A diagnostic is considered suppressible by a DiagnosticSuppressor if *all* of the following conditions are met:
+        ///     1. Diagnostic is not already suppressed in source via pragma/suppress message attribute.
+        ///     2. Diagnostic's <see cref="Diagnostic.DefaultSeverity"/> is not <see cref="DiagnosticSeverity.Error"/>.
+        ///     3. Diagnostic is not tagged with <see cref="WellKnownDiagnosticTags.NotConfigurable"/> custom tag.
+        /// </summary>
+        public ImmutableArray<Diagnostic> ReportedDiagnostics { get; }
```

Comment:
I would expect to just give them a set of diagnostics that they signed up for, but that odesn't sound like what this does.

Reply:
The implementation does match your expectation (see `getSuppressableDiagnostics` local function in AnalyzerDriver.cs). I have now made this explicit in the xml documentation comment.